### ISSUE TITLE
Devices view - what if we can't show a decent "last active" time?

### ISSUE THREAD

**Issue by rfk** [mozilla/fxa#190]
One of the acceptance criteria for the devices view [1] is that we show a "last active" timestamp for each device.  It turns out that tracking a last-accessed timestamp for millions of browsers is a Hard Problem, and when we tried to do this last time, we had to be back it out:

  https://github.com/mozilla/fxa-auth-server/issues/1044

We'll try again, but we should prepare a backup plan in case we just can't find a sensible way to track this timestamp reliably.

My proposal is:
- Update fxa-auth-server with a feature-flag to pref timestamp tracking on or off.  This will let us abort if db writes get too much, without having to roll back the entire train.
- When the feature is preffed off, have fxa-auth-server return an explicit `null` for the timestamps.  Right now it returns whatever timestamp we happened to manage to write to the database, which is pretty meaningless.
- In content-server, if we find that the timestamp is not available, we simply don't display the "last active: X" line in the UI.

An alternative might be to display the time at which the session was _created_, rather than the time it was last used.  I'm not sure precisely how we'd show that in the UI.

Thoughts @davismtl @ryanfeeley @vladikoff @philbooth?

[1] https://github.com/mozilla/fxa/blob/master/features/FxA-89-devices-view/README.md

**Comment by rfk** [comment]
Perhaps another way of putting this is: how much engineering effort are we willing to expend on getting the "last active" timestamp to work?

**Comment by davismtl** [comment]
- I think the feature flag makes sense (better safe than sorry)
- Null makes sense to me too
- Displaying nothing would prob be best UX

Let's determine how much effort we want to put in resolving it (if it doesn't work again) once we have engagement numbers with the feature in the wild.

**Comment by ryanfeeley** [comment]
@rfk @davismtl Because we don't offer geo-location information here, and most users aren't going to bother naming their devices, the date is pretty important (imagine a list of emails with the dates removed sorted in no apparent order).

From their POV we don't need to present a lot of information, so there will be an expectation that it is correct.

If we're unable to present a last active time, we should be transparent with the user:

* Last active: unknown (or more accurately, indeterminable)

Make sense?

**Comment by rfk** [comment]
> An alternative might be to display the time at which the session was created,
> rather than the time it was last used

Could an "active since" timestamp based on session login time, provide enough context to the user to be more useful than "unknown"?

**Comment by rfk** [comment]
> In content-server, if we find that the timestamp is not available, 
> we simply don't display the "last active: X" line in the UI.

@vladikoff just to sanity-check will the devices-view code that's landing in train-70 support missing/null lastAccessTime?

**Comment by ryanfeeley** [comment]
@rfk The one ramification of not being able to show the last active time for every device is likely deformatting the string.

i.e. I don't think we should show something like:

* Last active: 3 days ago
* Last active: 3 days ago
* Last active: 3 days ago
* Last active: 3 days ago
* First active: 29 days ago

It's not a good idea to use different labels for the same format.

I'd like to know what "materials" we're working with. In verbose terms, what can we say?

**Comment by rfk** [comment]
> The one ramification of not being able to show the last active time
> for every device is likely deformatting the string.

I think it's reasonable to assume that we can either show it for all devices, or for none of them.

**Comment by rfk** [comment]
The parameters I had in mind:
* We will always have the created-at timestamp for all devices.
* If the server is configured to track last-access-time, we will have it for all devices (to within 12hours granularity)
* If the server is configured not to track last-access-time, we will not report it for any devices

Also I should probably say more words about what's worrying me here, for context.

The last time we attempted this feature, we thought it had been successfully shipped.  The auth-server was deployed and chugging away tracking the last-access-time for devices.  It was only after it had spent a while in production that we realized, oh crap, the database write load keeps growing, we're going to have to back out this feature.

So my concern is that if this happens again, we might ship the devices view UI code and enable it for all users, and then after that discover that our last-access-time tracking is not sustainable and needs to be disabled.  If that happens, do we also need to backout the devices view code?  Or can we let it degrade gracefully to something that still provides value to the user?
